pkg/cloud/mesos/config.go

```diff
@@ -0,0 +1,79 @@
+package mesos
+
+import (
+	"io"
+	"time"
+
+	"code.google.com/p/gcfg"
+)
+
+const (
+	DefaultMesosMaster       = "localhost:5050"
+	DefaultHttpClientTimeout = time.Duration(10) * time.Second
+	DefaultStateCacheTTL     = time.Duration(5) * time.Second
+)
+
+var (
+	config *Config
+)
+
+func getConfig() *Config {
+	return config
+}
+
+func resetConfigToDefault() {
+	config = createDefaultConfig()
+}
+
+func init() {
+	resetConfigToDefault()
+}
+
+// Example Mesos cloud provider configuration file:
+//
+// [mesos-cloud]
+//  mesos-master        = leader.mesos:5050
+//	http-client-timeout = 500ms
+//	state-cache-ttl     = 1h
+
+type ConfigWrapper struct {
+	Mesos_Cloud Config
+}
+
+type Config struct {
+	MesosMaster            string          `gcfg:"mesos-master"`
+	MesosHttpClientTimeout WrappedDuration `gcfg:"http-client-timeout"`
+	StateCacheTTL          WrappedDuration `gcfg:"state-cache-ttl"`
+}
+
+type WrappedDuration struct {
```

Comment:
Two questions:

1. This will eventually (hopefully soon) be moved upstream.  I could put it in a common location at that time.
1. Wrapping every function we want to call on the underlying thing as in the linked example seems silly to me.  Is is really necessary?  If calling code want to invoke functions that we didn't think to wrap, they will have to go to the underlying member anyhow, or add more wrappers.  Then again, maybe I didn't understand what part of the linked code you were referring to.